### DISCUSSION TITLE
Rule for automatic overcurrent protection

### DISCUSSION THREAD

**Discussion by DJJaro18** [arendst/Tasmota#17102] [Q&A]
Hello.
Can someone write me cmd on tasmota firmware for " automatic circuit breaker"? wait time 10sec. MCU ESP8285.
Thanks.

**Comment by pkkrusty** [comment]
Not sure what device you are using to measure current, but in general, could look something like this:
```
Rule1
on Energy#Current>10 do ruletimer1 10 endon
on rules#timer=1 do power1 0 endon
on Energy#Current<10 do ruletimer1 0 endon
```
Enable one-shot with Rule1 5, then enable the rule with Rule1 1.
So if current goes over 10 amps, 10 second countdown will start. If at any point, current falls below 10 amps, timer is cancelled. This doesn't deal with resetting the power, or any follow-on logic, but it'll get you started.

**Comment by DJJaro18** [comment]
I have flashed SMTONOFF modular smart meter with switch. How enable rules? I try it... Thank you very mouch.

**Reply by DJJaro18** [reply]
I figured it out, but somehow it doesn't work for me.
08:22:11.028 CMD: Rule1 1
08:22:11.036 MQT: stat/tasmota_5B84C2/RESULT = {"Rule1":{"State":"ON","Once":"ON","StopOnError":"OFF","Length":121,"Free":390,"Rules":"on Energy#Current>10 do ruletimer1 10 endon on rules#timer=1 do power1 0 endon on Energy#Current<10 do ruletimer1 0 endon"}}

> "So if current goes over 10 amps, 10 second countdown will start. If at any point, current falls below 10 amps, timer is cancelled."

Rather, I would need it to turn off immediately on overcurrent and turn on again after 10 seconds, and if the overcurrent occurs again, turn off, wait, and turn on again, and the same thing over and over. Well thank you.

**Reply by DJJaro18** [reply]
Got it now, thanks for the kick :)
Rule1 
on Energy#Current>5 do power1 0 endon 
on Energy#Voltage<200 do ruletimer1 10 endon 
on rules#timer=1 do power1 1 endon
